pkg/apis/pipeline/v1beta1/task_validation.go

```diff
@@ -479,7 +479,9 @@ func validateStepVariables(step Step, prefix string, vars sets.String) *apis.Fie
 	errs := validateTaskVariable(step.Name, prefix, vars).ViaField("name")
 	errs = errs.Also(validateTaskVariable(step.Image, prefix, vars).ViaField("image"))
 	errs = errs.Also(validateTaskVariable(step.WorkingDir, prefix, vars).ViaField("workingDir"))
-	errs = errs.Also(validateTaskVariable(step.Script, prefix, vars).ViaField("script"))
```

Comment:
Wrapping this inside the if condition makes the test below fail. This was however required since parameters can be passed down implicitly.